### PR TITLE
Card constructor signature change, remove test package, finish IO

### PR DESCRIPTION
**Pull request by secminhr**
And some fixes also.

### CODE REVIEW

File: src/main/java/eod/IO/Input.java [hunk 1]
```diff
@@ -1,5 +1,20 @@
...
+
 public interface Input {
-    public void receiveMessage(String s);
+
+    //This method will send requesting message to user, waiting for its name
+    public String waitForPlayerName();
+
+    //This method will send requesting message to user, waiting for its response
+    //and return the desired card.
+    //Return null if player want to ended "Action stage(行動階段的英文不會)".
```
Comment (aoaaceai) [inline src/main/java/eod/IO/Input.java]: how about Action Phase?

File: src/main/java/eod/Player.java [hunk 2]
```diff
@@ -72,13 +90,18 @@ public void dropCards() {
     public void dropCards(int k) throws IllegalArgumentException {
         hand.randomlyDropCards(k);
     }
+
+    public void startAutoAttackInOrder() {
+        Gameboard board = game.getBoard();
+
```
Comment (aoaaceai) [inline src/main/java/eod/Player.java:97]: // TodO

File: src/main/java/eod/IO/Input.java [hunk 3]
```diff
@@ -1,5 +1,20 @@
 package eod.IO;
 
+import eod.card.abstraction.Card;
+
+import java.awt.*;
+
 public interface Input {
```
Comment (aoaaceai) [inline src/main/java/eod/IO/Input.java:8]: todo: connection with Player.selectObject, selectPosition, and selectCard
or maybe remove those methods.

File: src/main/java/eod/effect/Heal.java [hunk 4]
```diff
@@ -14,7 +14,7 @@ public Heal(Player player, int hp) {
         this.player = player;
     }
 
-    public void from(WarObject[] objects) {
+    public void to(WarObject[] objects) {
```
Comment (aoaaceai) [inline src/main/java/eod/effect/Heal.java:17]: // tOdO: add to(WarObject[] objects, int num) to heal multiple characters.

File: src/main/java/eod/IO/Input.java (hunk 1 above)
Comment (secminhr) [reply src/main/java/eod/IO/Input.java]: Changed.

File: src/main/java/eod/Player.java (hunk 2 above)
Comment (secminhr) [reply src/main/java/eod/Player.java:97]: added

File: src/main/java/eod/IO/Input.java (hunk 3 above)
Comment (secminhr) [reply src/main/java/eod/IO/Input.java:8]: Added

File: src/main/java/eod/effect/Heal.java (hunk 4 above)
Comment (secminhr) [reply src/main/java/eod/effect/Heal.java:17]: Added

File: src/main/java/eod/Player.java (hunk 2 above)
Comment (secminhr) [reply src/main/java/eod/Player.java:97]: And will be finished in next pr.

**Review by aoaaceai — APPROVED** [review]

**Pull request merged**